backend/routes/data.py

```diff
@@ -57,20 +112,41 @@ def add_data():
     audio_file.save(file_path.as_posix())
 
     app.logger.info(filename)
-
     try:
 
+        if isinstance(segmentations, str):
+            segmentations = json.loads(segmentations)
+        elif segmentations != []:
+            app.logger.error("Could not parse segmentations from ",
+                            type(segmentations))
+        annotations = []
+        for segment in segmentations:
+            validated = validate_segmentation(segment)
+            annotations.append(generate_segmentation(
+                data_id=None,
+                end_time=segment['end_time'],
+                start_time=segment['start_time'],
+                annotations=segment['annotations'],
+                transcription=segment['transcription'],
+            ))
+            if not validated:
```

Comment:
There's no point of checking validation after adding it to list of segmentations.

Reply:
Also, as discussed, if even one segmentation is invalid, we throw a 400 Bad Request then and there.

Reply:
Please throw a 400 here.

Reply:
The new code does throw 400 error.